tests/install/steps/test_expansion.py

```diff
@@ -0,0 +1,146 @@
+def deploy_node(version, k8s_client, name):
+    accept_ssh_key = [
+        'salt-ssh', '-i', name, 'test.ping', '--roster=kubernetes'
+    ]
+    pillar = {'bootstrap_id': 'bootstrap', 'node_name': name}
+    deploy = [
+        'salt-run', 'state.orch', 'metalk8s.orchestrate.deploy_new_node',
+        'saltenv=metalk8s-{}'.format(version),
+        'pillar={}'.format(json.dumps(pillar))
+    ]
+    run_salt_command(k8s_client, accept_ssh_key)
+    run_salt_command(k8s_client, deploy)
+
+
+# }}}
+# Then {{{
+
+@then(parsers.parse('node "{hostname}" is registered in Kubernetes'))
+def check_node_is_registered(k8s_client, hostname):
+    """Check if the given node is registered in Kubernetes."""
+    selector = 'metadata.name=={}'.format(hostname)
+    nodes = k8s_client.list_node(field_selector=selector).items
+    assert len(nodes) <= 1, 'too many nodes with hostname `{}`'.format(hostname)
+    assert len(nodes) == 1, '`{}` not registered in Kubernetes'.format(hostname)
+
+
+@then(parsers.parse('node "{node_name}" has the role master'))
+def check_master_role(k8s_client, node_name):
+    """Check if the given node meet the criteria to have the master role."""
+    namespace = 'kube-system'
+    selector  = 'spec.nodeName={}'.format(node_name)
+    pods = k8s_client.list_namespaced_pod(namespace, field_selector=selector)
+    pods_to_check = [
+        'kube-scheduler', 'kube-controller-manager', 'kube-apiserver'
+    ]
+    for pod in pods.items:
+        component = pod.metadata.labels.get('component')
+        if component in pods_to_check:
+            if pod.status.phase == 'Running':
+                pods_to_check.remove(component)
+    assert not pods_to_check, '{} should be up and running'.format(
+        ', '.join(pods_to_check)
+    )
+
+
+@then(parsers.parse('node "{node_name}" has the role etcd'))
+def check_etcd_role(k8s_client, node_name):
+    """Check if the given node meet the criteria to have the etcd role."""
+    namespace = 'kube-system'
+    field_selector = 'spec.nodeName={}'.format(node_name)
+    label_selector = 'component=etcd'
+    pods = k8s_client.list_namespaced_pod(
+        namespace, field_selector=field_selector, label_selector=label_selector
+    )
+    assert pods.items, 'etcd pod not found on node {}'.format(node_name)
+    assert pods.items[0].status.phase == 'Running', \
+        'etcd pod is not running on node {}'.format(node_name)
+
+    etcd_member_list = etcdctl(k8s_client, ['member', 'list'])
+    assert node_name in etcd_member_list, \
+        'node {} is not part of the etcd cluster'.format(node_name)
+
+
+# }}}
+# Helpers {{{
+
+def get_node_ip(hostname, ssh_config):
+    """Return the IP of the node `hostname`."""
+    infra_node = testinfra.get_host(hostname, ssh_config=ssh_config)
+    return infra_node.backend.client.get_transport().getpeername()[0]
+
+
+def get_node_manifest(metalk8s_version, node_ip):
+    """Return the YAML to declare a node with the specified IP."""
```

Comment:
Let's add a mention somewhere (either here or simply in the template name and the relevant function/`Given` names) that this is generating a **control-plane `Node`** (i.e. with the `master` and `etcd` roles). Future work could either make these roles customizable in the template (or in this function), or have different templates.

Reply:
Done